src/Nox.Types/Types/VatNumber/VatNumber.cs

```diff
@@ -1,9 +1,125 @@
+using System;
+using System.Text.RegularExpressions;
+
 namespace Nox.Types;
 
+/// <summary>
+/// Represents a Nox <see cref="VatNumber"/> type and value object.
+/// </summary>
+/// <remarks>Placeholder, needs to be implemented</remarks>
+public sealed class VatNumber : ValueObject<(string VatNumberValue, CountryCode2 CountryCode), VatNumber>
+{
+
+    #region VatNumber Regex
+
+    private static readonly Regex _vatNumberRegex = new(
```

Comment:
Would not be more simple to have a dictionary, key country code and value regex?

Reply:
Have updated to Dictionary